lib/pbench/server/auth/__init__.py

```diff
@@ -0,0 +1,435 @@
+    JWKS_URI: str = ""
+    LOGOUT_ENDPOINT: str = ""
+
+    def __init__(
+        self,
+        server_url: str,
+        client_id: str,
+        logger: logging.Logger,
+        realm_name: str = None,
+        client_secret_key: str = None,
+        verify: bool = True,
+        headers: Optional[Dict[str, str]] = None,
+    ):
+        self.server_url = server_url
+        self.client_id = client_id
+        self.client_secret_key = client_secret_key
+        self.realm_name = realm_name
+        self.logger = logger
+        self.headers = (
+            CaseInsensitiveDict()
+            if headers is None
+            else CaseInsensitiveDict().update(headers)
+        )
+        self.verify = verify
+        self.connection = requests.session()
+        self.set_well_known_endpoints()
+
+    def get_header_param(self, key: str) -> str:
+        """
+        Return a specific header parameter value.
+        Args:
+            key: header key
+        Returns:
+             Header parameters value.
+        """
+        return self.headers.get(key)
+
+    def add_header_param(self, key: str, value: str):
+        """
+        Add a single parameter inside the header.
+        Args:
+            key: Header parameters key.
+            value: Value to be added for the key.
+        """
+        self.headers[key] = value
+
+    def del_param_headers(self, key: str):
+        """Remove a specific header parameter.
+        Args:
+            Key to delete from the headers.
+        """
+        del self.headers[key]
+
+    def set_well_known_endpoints(self):
+        """Sets the well-known configuration endpoints for the current oidc client.
+        This includes common useful endpoints for decoding tokens, getting user
+        information etc.
+        """
+        well_known_endpoint = "/.well-known/openid-configuration"
+        if self.realm_name:
+            well_known_uri = f"{self.server_url}{self.realm_name}{well_known_endpoint}"
+        else:
+            well_known_uri = f"{self.server_url}{well_known_endpoint}"
+        endpoints_json = self._get(well_known_uri).json()
+        try:
+            OpenIDClient.AUTHORIZATION_ENDPOINT = endpoints_json[
+                "authorization_endpoint"
+            ]
+            OpenIDClient.TOKEN_ENDPOINT = endpoints_json["token_endpoint"]
+            OpenIDClient.USERINFO_ENDPOINT = endpoints_json["userinfo_endpoint"]
+            OpenIDClient.REVOCATION_ENDPOINT = endpoints_json["revocation_endpoint"]
+            OpenIDClient.JWKS_ENDPOINT = endpoints_json["jwks_uri"]
+            if "end_session_endpoint" in endpoints_json:
+                OpenIDClient.LOGOUT_ENDPOINT = endpoints_json["end_session_endpoint"]
+        except KeyError as e:
+            self.logger.exception("{}", str(e))
+            raise
+
+    def get_oidc_public_key(self, token: str):
+        """
+        Returns the Oidc client public key that can be used for decoding tokens offline.
```

Comment:
I suspect that "Oidc" should be spelled in solid-caps.

Reply:
There are other instances of "oidc" in comments and messages which you might want to up-case, as well.